src/components/Tooltip/tooltipPropTypes.js

```diff
@@ -38,6 +38,9 @@ const propTypes = {
 
     /** Render custom content inside the tooltip. Note: This cannot be used together with the text props. */
     renderTooltipContent: PropTypes.func,
+
+    /** Unique key of renderTooltipContent to rerender the tooltip when one of the key changes */
+    renderTooltipContentKey: PropTypes.arrayOf(PropTypes.any),
```

Comment:
I see your suggestion to do
```js
renderTooltipContentKey: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.bool,
])),
```

and that would be fine, but is there any reason we can't simply restrict to strings for now?
```suggestion
    renderTooltipContentKey: PropTypes.arrayOf(PropTypes.string),
```
Looks like that's all that's being passed in from what I can see